presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestDistributedQueries.java

```diff
@@ -243,6 +246,45 @@ public void testCreateTable()
         assertFalse(getQueryRunner().tableExists(getSession(), "test_create_like"));
     }
 
+    @Test
+    public void testNoneAutoCommitTransactionWithFailAndRollback()
+    {
+        assertUpdate("create table test_non_autocommit_table(a int, b varchar)");
+        Session session = getQueryRunner().getDefaultSession();
+        String defaultCatalog = session.getCatalog().get();
+        transaction(getQueryRunner().getTransactionManager(), getQueryRunner().getAccessControl())
+                .execute(Session.builder(session)
+                                .setIdentity(new Identity("admin",
+                                        Optional.empty(),
+                                        ImmutableMap.of(defaultCatalog, new SelectedRole(ROLE, Optional.of("admin"))),
+                                        ImmutableMap.of(),
+                                        ImmutableMap.of(),
+                                        Optional.empty(),
+                                        Optional.empty()))
+                                .build(),
+                        txnSession -> {
+                            // simulate failure of SQL statement execution
+                            assertQueryFails(txnSession, "SELECT fail('forced failure')", "forced failure");
+
+                            // cannot execute any SQLs except `rollback` in current session
+                            assertQueryFails(txnSession, "select count(*) from test_non_autocommit_table", "Current transaction is aborted, commands ignored until end of transaction block");
+                            assertQueryFails(txnSession, "show tables", "Current transaction is aborted, commands ignored until end of transaction block");
+                            assertQueryFails(txnSession, "insert into test_non_autocommit_table values(1, '1001')", "Current transaction is aborted, commands ignored until end of transaction block");
+                            assertQueryFails(txnSession, "create table test_table(a int, b varchar)", "Current transaction is aborted, commands ignored until end of transaction block");
+
+                            // execute `rollback` successfully
+                            assertUpdate(txnSession, "rollback");
+                        });
+
+        assertQuery("select count(*) from test_non_autocommit_table", "values(0)");
+        try {
+            assertUpdate("drop table if exists test_non_autocommit_table");
+        }
+        catch (Exception e) {
+            // ignored for connector compatibility
```

Comment:
What does "for connector compatibility imply" here? Do some connectors behave differently?